packages/zoe/src/contracts/autoswap.js

```diff
@@ -94,23 +93,16 @@ const start = async zcf => {
   };
 
   function consummate(tradeAmountIn, tradeAmountOut, swapSeat) {
-    trade(
-      zcf,
-      {
-        seat: poolSeat,
-        gains: {
-          [getPoolKeyword(tradeAmountIn.brand)]: tradeAmountIn,
-        },
-        losses: {
-          [getPoolKeyword(tradeAmountOut.brand)]: tradeAmountOut,
-        },
-      },
-      {
-        seat: swapSeat,
-        gains: { Out: tradeAmountOut },
-        losses: { In: tradeAmountIn },
-      },
-    );
+    poolSeat.incrementBy({
+      [getPoolKeyword(tradeAmountIn.brand)]: tradeAmountIn,
+    });
+    swapSeat.decrementBy({ In: tradeAmountIn });
+
+    swapSeat.incrementBy({ Out: tradeAmountOut });
+    poolSeat.decrementBy({
+      [getPoolKeyword(tradeAmountOut.brand)]: tradeAmountOut,
+    });
+    zcf.reallocate(swapSeat, poolSeat);
```

Comment:
The pool doesn't have any `want` so this case can't use Swap or SwapExact, and I think this particular usage isn't extractable into something shorter, since the keywords have to be specified as they are different for each seat. I think this is both legible and pretty short.

Reply:
Satisfied. Resolving.